src/protocols/tcp6.rs

```diff
@@ -0,0 +1,178 @@
+//! Transmission Control Protocol version 6
+//!
+//! It provides services to send and receive data stream.
+
+use crate::efi::{Boolean, Ipv6Address};
+use crate::protocols::{ip6, managed_network, simple_network};
+
+pub const PROTOCOL_GUID: crate::base::Guid = crate::base::Guid::from_fields(
+    0x46e44855,
+    0xbd60,
+    0x4ab7,
+    0xab,
+    0x0d,
+    &[0xa6, 0x79, 0xb9, 0x44, 0x7d, 0x77],
+);
+
+#[repr(C)]
+pub struct AccessPoint {
+    pub station_address: Ipv6Address,
+    pub station_port: u16,
+    pub remote_address: Ipv6Address,
+    pub remote_port: u16,
+    pub active_flag: Boolean,
+}
+
+#[repr(C)]
+pub struct Tcp6Option {
```

Comment:
I dropped the `Tcp6` prefix, as we already have it in the module name. We can safely use `r#Option` to avoid the naming conflict.